File: tests/base/test_dependencies.py
import pytest
import sys
import os

def test_node_available():
    """Test that Node.js is available in the environment."""
    import subprocess
    result = subprocess.run(['node', '--version'], capture_output=True, text=True)
    assert result.returncode == 0, "Node.js should be available"
    assert result.stdout.strip().startswith('v'), "Node.js version should be reported"

def test_npm_available():
    """Test that npm is available in the environment."""
    import subprocess
    result = subprocess.run(['npm', '--version'], capture_output=True, text=True)
    assert result.returncode == 0, "npm should be available"

def test_typescript_compiles():
    """Test that TypeScript can compile the project."""
    import subprocess
    repo_root = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
    result = subprocess.run(['npm', 'run', 'build'], cwd=repo_root, capture_output=True, text=True)
    assert result.returncode == 0, f"TypeScript compilation should succeed: {result.stderr}"

def test_python_available():
    """Test that Python is available in the environment."""
    assert sys.version_info.major >= 3, "Python 3 should be available"

def test_pytest_available():
    """Test that pytest is available."""
    import pytest
    assert hasattr(pytest, '__version__'), "pytest should be installed"

File: tests/base/test_dependencies.ts
import { test } from 'node:test';
import assert from 'node:assert';
import { execSync } from 'node:child_process';

// Use process.cwd() since tests run from repository root
const repoRoot = process.cwd();

test('Node.js is available', () => {
  const result = execSync('node --version', { encoding: 'utf-8' });
  assert(result.trim().startsWith('v'), 'Node.js version should be reported');
});

test('npm is available', () => {
  execSync('npm --version', { encoding: 'utf-8' });
  // If no exception is thrown, npm is available
});

test('TypeScript can compile the project', () => {
  try {
    execSync('npm run build', { 
      cwd: repoRoot, 
      encoding: 'utf-8',
      stdio: 'pipe'
    });
  } catch (error: any) {
    assert.fail(`TypeScript compilation should succeed: ${error.stderr || error.message}`);
  }
});

test('Python is available', () => {
  try {
    const result = execSync('python3 --version', { encoding: 'utf-8' });
    const versionMatch = result.match(/Python (\d+)/);
    assert(versionMatch !== null, 'Python version should be reported');
    const majorVersion = parseInt(versionMatch![1], 10);
    assert(majorVersion >= 3, 'Python 3 should be available');
  } catch (error) {
    assert.fail('Python 3 should be available');
  }
});

test('pytest is available', () => {
  try {
    execSync('python3 -m pytest --version', { encoding: 'utf-8', stdio: 'pipe' });
    // If no exception is thrown, pytest is available
  } catch (error) {
    assert.fail('pytest should be installed');
  }
});

